File: app/backend/app/schemas/scan.py
from typing import List

from pydantic import BaseModel


class ScanResponse(BaseModel):
    waste_type: str
    category: str
    recycling_category: str
    confidence: float
    eco_tip: str
    recycling_advice: str
    preparation_steps: List[str]
    recyclable: bool
    earned_points: int
    scan_reward: int
    task_rewards: int
    daily_task_rewards: int
    weekly_task_rewards: int
    auto_claimed_task_ids: List[str]
    total_reward: int
    new_total_points: int

File: app/types/scan.ts
// Types for Scan Waste functionality

export interface RecyclingCenter {
  name: string;
  address: string;
  acceptedWasteTypes: string[];
  latitude: number;
  longitude: number;
}

export interface WasteDetectionResult {
  waste_type: string;
  category: string;
  recycling_category: string;
  confidence: number;
  eco_tip: string;
  recycling_advice: string;
  preparation_steps: string[];
  recyclable: boolean;
}

export interface ScanResponse {
  waste_type: string;
  category: string;
  recycling_category: string;
  confidence: number;
  eco_tip: string;
  recycling_advice: string;
  preparation_steps: string[];
  recyclable: boolean;
  earned_points: number;
  scan_reward: number;
  task_rewards: number;
  daily_task_rewards: number;
  weekly_task_rewards: number;
  auto_claimed_task_ids: string[];
  total_reward: number;
  new_total_points: number;
}

export interface ScanState {
  selectedImage: string | null;
  isScanning: boolean;
  scanResult: WasteDetectionResult | null;
  scanResponse: ScanResponse | null;
  loadingMessage: string;
  error: string | null;
  recyclingCenter: RecyclingCenter | null;
}

export type LoadingState =
  "idle" | "uploading" | "analyzing" | "generating" | "complete" | "error";
